torch/signal/windows/__init__.py

```diff
@@ -0,0 +1,159 @@
+import warnings
+
+from .windows import cosine, exponential, gaussian
+
+from torch._torch_docs import factory_common_args
+
+__all__ = [
+    'cosine',
+    'exponential',
+    'gaussian',
+]
+
+
+def _add_docstr(function, docstr):
+    function.__doc__ = docstr
+
+
+_add_docstr(
```

Comment:
No need to do this. You can simply put the docs inside the function as you were doing. My suggestion was just to use the `factory_common_args` as you are now doing :)

Note that we add the docs in the torch functions after the fact, because the Python bindings are codgen'd.

Reply:
Do you mean something like this?

```python
def function(a: int, b: int, c: int = 0):
    r"""
    A function
    
    Args:
       a (int): parameter a
       b (int): parameter b

    Keyword args:
       {c}
    """.format(
          **kwargs
     )

     # function code
```

'cause I don't think it works.

Reply:
ugh. Then, what about a decorator like this, that still keeps the function and the docs close together?

https://stackoverflow.com/questions/32152071/can-i-format-the-docstring-in-python3

Reply:
Ok, now that I had time to look at it, it's a bit more complex because our docstrings support LaTeX. I'm trying to think of a pretty and friendly way to support this in a decorator, we need to apply `.format()` to a specific string and not the whole docstring.

Reply:
Done!